### PR TITLE
refactor: remove async module usage

### PR DESCRIPTION
**Pull request by vijay-qlogic**
Fixes #[2883](https://github.com/googleapis/google-cloud-node/issues/2883) partially for nodejs-datastore (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: system-test/datastore.ts [hunk 1]
```diff
@@ -15,41 +15,39 @@
...
-          callback(err);
-          return;
-        }
-
-        const keys = entities.map(entity => {
-          return entity[datastore.KEY];
-        });
-
-        datastore.delete(keys, callback);
+      const data = await datastore.runQuery(query);
```
Comment (JustinBeckwith) [inline system-test/datastore.ts]: collapse to `const [entities] = await datastore.runQuery(query);`

File: system-test/datastore.ts [hunk 2]
```diff
@@ -15,41 +15,39 @@
...
     testKinds.push(keyObject.kind);
     return keyObject;
   };
 
-  after(done => {
-    function deleteEntities(kind, callback) {
-      const query = datastore.createQuery(kind).select('__key__');
+  after(async () => {
+    async function deleteEntities(kind) {
+      const query = await datastore.createQuery(kind).select('__key__');
```
Comment (JustinBeckwith) [inline system-test/datastore.ts]: This doesn't need to be await'ed.  `createQuery` is a synchronous method.

File: system-test/datastore.ts [hunk 3]
```diff
@@ -15,41 +15,39 @@
...
+      const entities = data[0];
+      const keys = entities.map(entity => {
+        return entity[datastore.KEY];
       });
+
+      await datastore.delete(keys);
     }
 
-    async.each(testKinds, deleteEntities, done);
+    let promises: Array<{}> = [];
```
Comment (JustinBeckwith) [inline system-test/datastore.ts]: No need for the loop. 

`const promises = testKinds.map(kind => deleteEntities(kind))`

File: system-test/datastore.ts [hunk 4]
```diff
@@ -861,34 +859,32 @@ describe('Datastore', () => {
               },
             ]);
 
-            transaction.commit(err => {
+            transaction.commit(async err => {
```
Comment (JustinBeckwith) [inline system-test/datastore.ts]: Having an async callback method is super weird.  I believe `transaction.commit` is an async function.  Can you just `await` the result of that?

File: system-test/datastore.ts [hunk 5]
```diff
@@ -861,34 +859,32 @@ describe('Datastore', () => {
...
-                  callback => {
-                    datastore.get(deleteKey, (err, entity) => {
-                      assert.ifError(err);
-                      assert.strictEqual(typeof entity, 'undefined');
-                      callback();
-                    });
-                  },
+              await Promise.all([
+                // Deletes the key that is in the deletion queue.
+                callback => {
```
Comment (JustinBeckwith) [inline system-test/datastore.ts]: Mixing callback and promise based code is not good :)  These shouldn't be callbacks, and we should be using awaits throughout here.

File: system-test/datastore.ts [hunk 6]
```diff
@@ -830,69 +824,51 @@ describe('Datastore', () => {
...
-                  callback => {
-                    datastore.get(deleteKey, (err, entity) => {
-                      assert.ifError(err);
-                      assert.strictEqual(typeof entity, 'undefined');
-                      callback();
-                    });
-                  },
+      await Promise.all([
+        // Deletes the key that is in the deletion queue.
+        datastore.get(deleteKey, (err, entity) => {
```
Comment (JustinBeckwith) [inline system-test/datastore.ts]: Mixing callbacks and promises. Use a `.then` to do the assertion.

Comment (vijay-qlogic) [reply system-test/datastore.ts]: Done! https://github.com/googleapis/nodejs-datastore/pull/272/commits/c120a83d74bdb5a4c96e361ab7cf40b4684839bc

**Comment by JustinBeckwith** [comment]
👋 we could use a fixy on the merge conflicts :)

**Pull request merged**